modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/authenticator/JaasAuthenticator.java

```diff
@@ -40,30 +44,48 @@
 public class JaasAuthenticator implements Authenticator {
 
     @Override
-    public void initialize(StartupContext startupContext) throws Exception {
-        UserStoreManager userStoreManager = startupContext.getService(UserStoreManager.class);
-        BrokerConfigProvider configProvider = startupContext.getService(BrokerConfigProvider.class);
-        BrokerAuthConfiguration brokerAuthConfiguration = configProvider
-                .getConfigurationObject(BrokerAuthConfiguration.NAMESPACE, BrokerAuthConfiguration.class);
-        BrokerAuthConfiguration.JaasConfiguration jaasConf = brokerAuthConfiguration.getAuthentication().getJaas();
+    public void initialize(StartupContext startupContext, Map<String, Object> properties) throws Exception {
+
         String jaasConfigPath = System.getProperty(BrokerAuthConstants.SYSTEM_PARAM_JAAS_CONFIG);
+        UserStoreConnector userStoreConnector = new FileBasedUserStoreConnector();
+        userStoreConnector.initialize(startupContext);
+        startupContext.registerService(UserStoreConnector.class, userStoreConnector);
```

Comment:
Do we need to register the UserStoreConnector? I don't see a place where we do a get on this service.